server/router/src/test/java/org/infinispan/server/router/performance/RouterPerf.java

```diff
@@ -0,0 +1,121 @@
+package org.infinispan.server.router.performance;
+
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.server.hotrod.HotRodServer;
+import org.infinispan.server.router.MultiTenantRouter;
+import org.infinispan.server.router.routes.Route;
+import org.infinispan.server.router.routes.RouteDestination;
+import org.infinispan.server.router.routes.RouteSource;
+import org.junit.Test;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This class is responsible for performance tests against the router. It tests 3 configurations:
+ * <ul>
+ *     <li>No SSL at all</li>
+ *     <li>HotRod with SSL only</li>
+ *     <li>Multi tenant router with SSL+SNI</li>
+ * </ul>
+ *
+ * <p>
+ *     Note that this class is not triggered by Surefire by default (it doesn't end with "test"). We want to do performance
+ *     test on demand only.
+ * </p>
+ */
+public class RouterPerf {
+
+    private static final int MEASUREMENT_ITERATIONS_COUNT = 1;
+    private static final int WARMUP_ITERATIONS_COUNT = 2;
+
+    @Test
+    public void performRouterBenchmark() throws Exception {
+        Options opt = new OptionsBuilder()
+                .include(this.getClass().getName() + ".*")
+                .mode(Mode.AverageTime)
+                .mode(Mode.SingleShotTime)
+                .timeUnit(TimeUnit.MILLISECONDS)
+                .warmupIterations(WARMUP_ITERATIONS_COUNT)
+                .measurementIterations(MEASUREMENT_ITERATIONS_COUNT)
```

Comment:
I'm sorry, I decreased the number of iteration because when I performed the test (31 iterations) it took me about ~2h. I though that no one will be so patient :)

But you are absolutely right - I should have left 31.